internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -214,7 +212,17 @@ func (c *collection) runWrites(ctx context.Context, actions []*driver.Action) er
 		ClientRequestToken: aws.String(driver.UniqueString()),
 		TransactItems:      tws,
 	})
-	return err
+	if err != nil {
+		return err
+	}
+	for i, a := range actions {
+		if a.Kind == driver.Create {
+			if _, err := a.Doc.GetField(c.partitionKey); err != nil && gcerrors.Code(err) == gcerrors.NotFound {
+				actions[i].Doc.SetField(c.partitionKey, tws[i].Put.Item[c.partitionKey])
```

Comment:
`tws[i].Put.Item[c.partitionKey]` is an `AttributeValue`, I think. You have to extract the `S` field.

Reply:
Can you also add something to the driver test for this case that checks that the generated partition key is a string, so this bug would have been caught? Either in this PR, or add a TODO.

Reply:
Good catch, fixed.

There is already a TODO in `testCreate` that I currently comment out the `unnamed` test case. Filed #1857